xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3646,6 +3644,13 @@ void CVideoPlayer::SetSpeed(float speed)
     return;
   
   m_newPlaySpeed = speed * DVD_PLAYSPEED_NORMAL;
+  if (m_newPlaySpeed != m_playSpeed)
+  {
+      if (m_newPlaySpeed == DVD_PLAYSPEED_NORMAL)
+          m_callback.OnPlayBackResumed();
+      else if (m_newPlaySpeed == DVD_PLAYSPEED_PAUSE)
+          m_callback.OnPlayBackPaused();
+  }
   SetPlaySpeed(speed * DVD_PLAYSPEED_NORMAL);
```

Comment:
I could not find my initial comment to @Razzeee anymore, so again: This is an intermediate fix and it does not fix the real issue that is the blocking callback interface. Because it is blocking VideoPlayer's internal thread can't call it. @garbear SetPlaySpeed is public but must not block because it is also called by the internal thread.

the current state: it works somehow and continues to have issues as it has always had. An external call to Pause() or SetSpeed just triggers an asynchronous command to VideoPlayer. The real action happens later. As a result the callback is just a guess what player might do later on. It can be wrong.